heareval/tasks/util/luigi.py

```diff
@@ -176,15 +179,22 @@ def get_metadata(self):
             header=None,
             names=PROCESSMETADATACOLS,
         )[["filename_hash", "slug", "relpath", "partition"]]
+        # Since event detection metadata will have duplicates, we de-dup
+        # TODO: We might consider different choices of subset
+        metadata = metadata.sort_values(by="filename_hash").drop_duplicates(
+            subset="filename_hash", ignore_index=True
+        )
         return metadata
 
     def run(self):
 
         process_metadata = self.get_metadata()
         # Subsample the files based on max files per corpus.
         # The filename hash is used here
-        # This task can also be done in the configprocessmetadata as that will give
-        # freedom to stratify the selection on some criterion?
+        # This task can also be done in ConfigProcessMetadata as that will give
+        # freedom to stratify the selection on some criterion? That seems kinda
+        # fiddly, if we want fancy stratification we should discuss that
+        # separately.
```

Comment:
@jorshi @khumairraj I think we should talk this over.

Reply:
Ya -- we aren't doing any stratification presently. If we are subsampling severely we might need to think about it. Not sure exactly where to do that -- especially with sed tasks.

Reply:
https://github.com/neuralaudio/hear2021-eval-kit/issues/86

Reply:
Yes. Since we were not doing any fancy preprocessing, I suggested to move this to the `ConfigProcessMetadata`. For speecheval we are subsampling the data on the basis of the id of the user recording the speech(handled by the filehash as discussed before). We can decide on keeping or moving this.